src/convert.rs

```diff
@@ -0,0 +1,42 @@
+pub trait CAPIConverter {
```

Comment:
I'm still not sure this trait gets it right.  I think we need to play with strings to find a trait that does.  Which is to say, this isn't really any different than what was here before, and I believe it will have to evolve again in the near future.

Reply:
I think I said this in slack already but we will need to distinguish between types which are trivially convertible (as each of this examples is) and types which are not (likely string).  For the trivially convertible cases, we don't want any functions at all, just a marker - and then for the non-trivial cases we will end up with implementations that are more complicated than this.

Reply:
The conversion stuff isn't any different for sure. The issue was I focusing on was removing the Datatype dependence of the conversion. With get_fill_value I ran smack into the issue when I realized that DomainType was wrong because u64 would only have one associated Datatype via DomainType::DATATTYPE which was wrong.

Reply:
Ah that makes sense.  Yeah multiple datatypes backed by u64.  Got it.